Library/Homebrew/cmd/--version.rb

```diff
@@ -19,7 +19,7 @@ def __version_args
   def __version
     __version_args.parse
 
-    odie "This command does not take arguments." if ARGV.any?
+    raise "This command does not take arguments" if args.remaining.any?
```

Comment:
I wonder if adding a new exception for this specific case would be a good idea (or even an argument to #parse`) ?

Reply:
I took a shot at implementing this in `cli/parser`; have a look.